### ISSUE TITLE
Move --supersede out of list and into separate command

### ISSUE THREAD

**Issue by jberry-suse** [openSUSE/osc-plugin-factory#738]
It always seemed odd that --supersede was part of list and per email discussion it seems I am not alone. Perhaps a supersede command would be better.

**Comment by nilxam** [comment]
then just run staging list without --supersede parameter, the problem is only happened on Leap that just because it doesn't have devel project sets, list --supersede just superseding the staged request if it has newer request in backlog, otherwise it will be potentially staging two requests in two staging project but from same source package.

**Comment by jberry-suse** [comment]
I believe that's related to #739. I agree that could end up with two packages and I am not sure what @lnussel intended in that case. This issue was to track breaking out `--supersede` as a separate command since it does not really have anything to do with `list`.

**Comment by nilxam** [comment]
I thought the origin reason is about --supersede does not work well for Leap process, maybe I read it wrong from mail discussion, but still, I don't like this idea https://github.com/openSUSE/osc-plugin-factory/pull/753, split out supersede to another command is unnecessary IMO.

With this change, we need to run `osc staging list` to get backlog list first, and run `osc staging supersede XX` then, afterward we need to `osc staging list` to see the current backlog and ensure the request was superseded, it's superfluous. It was part of list command is due to we want to see the proper list what we looking for, before superseding or after superseding.

**Comment by lnussel** [comment]
if you always unconditionally call list --supersede on TW it should be the default mode of operation I guess. However, to me it looked like something you have to to consciously, esp when the package submitter for whatever reason did not supersede the request himself.
For Leap the blind --supersede is undesirable to harmful. So there the decision needs to be explicit always.

How do we satisfy both use cases? :-) Config file somewhere?

**Comment by lnussel** [comment]
in fact I think obs lacks  way to lock requests so package maintainers cannot supersede themselves and it's up to us to do that. That would prevent invalidating stagings.

**Comment by DimStar77** [comment]
I don't think we as Release Managers have all the info to know why a contributor supersedes the request. He might have very valid reasons, like a regression spotted. Even though staging could turn out green - it would be damaging to check it in and disallow the contributor to supersede his requests

**Comment by DimStar77** [comment]
> How do we satisfy both use cases? :-) Config file somewhere?

config file is an option there - the big diff for Leap vs TW is definitively that TW has a set of acceptable source per package, which Leap does not have (so supersede in TW is 'normal' and 'wanted', in Leap anybody can basically disrupt the service

OTOH, I am so used to calling 'osc staging list --supersede' that this is not even considered a burden to me :)

**Comment by nilxam** [comment]
For Leap, you don't need to call list --supersede unconditionally, you can just call `osc staging list` without --supersede doesn't? and then decline the request you don;t like or supersede it.

**Comment by lnussel** [comment]
I basically never call list --supersede as I don't know what would happen. It would randomly choose eg between a Factory package submission vs one from update crawler.

**Comment by DimStar77** [comment]
So leaving it as-is sounds like the sanest approach:

for Lap, you use osc staging list, for TW we use osc staging list --supersede.

**Comment by jberry-suse** [comment]
The reasoning behind breaking out supersede from list was that it currently acts as a somewhat arbitrary side-effect of running list. More about breaking out responsibilities of commands.

Given how automatic supersede is for TW it is a perfect candidate for being run automatically which would make it somewhat of a moot topic. No reason to wait for someone to run the list command to perform the supersede.

The behavior in Leap should likely still be figured out. If anything it seems like a couple cases should be handled in Leap.

- two requests from different source projects with the same diff: preference order (otherwise, it really does not seem to matter)
- two requests with different diff
  - same source project: supersede as normal
  - different source project: leave for manual review

Essentially, the same as TW except for the last case specific to Leap.

**Comment by DimStar77** [comment]
For TW, the staging master still needs to retain the option though to have supersede implicit done when listing the backlog; there is simply never the case that we want to see the backlog without replacing what is already lying around.
